cudamapper/src/chainer_utils.cuh

```diff
@@ -0,0 +1,149 @@
+/*
+* Copyright 2020 NVIDIA CORPORATION.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#pragma once
+
+#include <claraparabricks/genomeworks/cudamapper/types.hpp>
+#include <claraparabricks/genomeworks/utils/device_buffer.hpp>
+
+namespace claraparabricks
+{
+
+namespace genomeworks
+{
+
+namespace cudamapper
+{
+namespace chainerutils
+{
+
+struct OverlapToNumResiduesOp
+{
+    __device__ __forceinline__ int32_t operator()(const Overlap& overlap) const
+    {
+        return overlap.num_residues_;
+    }
+};
+
+///
+/// \brief Create an overlap from the first and last anchor in the chain and
+/// the total number of anchors in the chain.
+///
+/// \param start The first anchor in the chain.
+/// \param end The last anchor in the chain.
+/// \param num_anchors The total number of anchors in the chain.
+__host__ __device__ Overlap create_simple_overlap(const Anchor& start,
```

Comment:
This function is only called by `backtrace_anchors_to_overlaps()`. I understand that it cannot be in an anonymous namespace in order be able to call it form unittest, but it could be moved to `details` namespace in order to signal the users that they do not have to think about it

Reply:
I think creating an overlap from two anchors and a number of residues is useful and could be a user-facing API function, but the function could be better named.

Reply:
I agree. That's why I was suggesting moving it closed to `Overlap`'s definition: https://github.com/clara-parabricks/GenomeWorks/pull/590#discussion_r510250360

Reply:
I see. I've renamed it to `create_overlap` in c72e5c96.